crates/types/src/lib.rs

```diff
@@ -232,6 +232,8 @@ pub struct HotShotConfig<KEY: SignatureKey> {
     pub start_voting_time: u64,
     /// Unix time in seconds at which we stop voting on an upgrade. To prevent voting on an upgrade, set stop_voting_time <= start_voting_time.
     pub stop_voting_time: u64,
+    /// Number of blocks in an epoch, zero means there are no epochs
+    pub epoch_height: u64,
```

Comment:
this might be personal preference, but I feel like this is a protocol-level constant and might be better as an associated constant in the `NodeType` trait (rather than a configuration value)

Reply:
yeah this one is weird, view timeout is also a parameter of the protocol but is specified in this config.  If we were to move this to associated constant, we need to do so for all protocol values (which is at least a few of these values)

Reply:
I guess the distinction in my mind was:

nodes could set different view timeouts, round_start_delay etc (within reason) and consensus would probably be fine. nodes could even update it dynamically between views (within reason) and I don't think anything would break

but every single node needs to have exactly the same epoch height, forever, and if we change this at all without an upgrade consensus would break